### PR TITLE
Servicing updates

### PR DESCRIPTION
**Pull request by KSchlobohm**
Provides additional configurations for security settings.

### CODE REVIEW

File: infra/core/compute/postDeploymentScript/post-deployment.sh [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #!/bin/bash
 
 # install AZD
-curl -fsSL https://aka.ms/install-azd.sh | sudo bash
+curl -fsSL https://aka.ms/install-azd.sh | bash -s -- -- version 1.9.5
```
Comment (KSchlobohm) [inline infra/core/compute/postDeploymentScript/post-deployment.sh:4]: add comment - keep in sync with version in other files

File: infra/modules/shared-frontdoor.bicep [hunk 2]
```diff
@@ -72,6 +74,39 @@ module frontDoor '../core/security/front-door-with-waf.bicep' = {
       { name: 'Microsoft_DefaultRuleSet', version: '2.1' }
       { name: 'Microsoft_BotManagerRuleSet', version: '1.0' }
     ] : []
+
+    customRules: {
+      rules: [{
+        name: 'RateLimitRule2'
```
Comment (KSchlobohm) [inline infra/modules/shared-frontdoor.bicep:80]: RateLimitRule1 not 2

File: infra/core/database/create-sql-user-and-role.bicep [hunk 3]
```diff
@@ -59,7 +59,7 @@ resource createSqlUserAndRole 'Microsoft.Resources/deploymentScripts@2020-10-01'
       forceUpdateTag: uniqueScriptId
       azPowerShellVersion: '7.4'
       retentionInterval: 'PT1H'
-      cleanupPreference: 'OnExpiration'
+      cleanupPreference: 'OnSuccess'
```
Comment (twsouthwick) [inline infra/core/database/create-sql-user-and-role.bicep:62]: can you make a comment as to why this is necessary?

File: infra/core/security/front-door-with-waf.bicep [hunk 4]
```diff
@@ -102,7 +148,7 @@ resource frontDoorEndpoint 'Microsoft.Cdn/profiles/afdEndpoints@2023-05-01' = {
   }
 }
 
-resource wafPolicy 'Microsoft.Network/FrontDoorWebApplicationFirewallPolicies@2022-05-01' = {
+resource wafPolicy 'Microsoft.Network/frontdoorwebapplicationfirewallpolicies@2024-02-01' = {
```
Comment (twsouthwick) [inline infra/core/security/front-door-with-waf.bicep:151]: I take it there's no case sensitivity here? the all lowercase feels weird, though

**Pull request merged**